### PR TITLE
Avoid copy of data when create Constant from HostTensor

### PR DESCRIPTION
**Pull request by ilyachur**

### CODE REVIEW

File: ngraph/core/src/op/constant.cpp [hunk 1]
```diff
@@ -50,10 +50,24 @@ string to_cpp_string(T value)
...
+            static_cast<char*>(hostTensor->get_data_ptr()), tensor->get_size_in_bytes(), tensor);
+    }
+    else
+    {
+        constructor_validate_and_infer_types();
+        allocate_buffer();
+        tensor->read(get_data_ptr_nc(), tensor->get_size_in_bytes());
+        m_all_elements_bitwise_identical = are_all_data_elements_bitwise_identical();
+    }
+    constructor_validate_and_infer_types();
```
Comment (jane-intel) [inline ngraph/core/src/op/constant.cpp:70]: why we do it twice for not HostTensors?

Comment (ilyachur) [reply ngraph/core/src/op/constant.cpp:70]: Because method constructor_validate_and_infer_types() checks shapes and types and allocates buffer.

**Comment by ilya-lavrenov** [comment]
is it still valid?

**Comment by ilyachur** [comment]
I faced with some issues with LPT transformation. We have a ticket to complete this PR in order to improve the memory consumption but at the current moment looks like improvements won't so big.
So I will close this PR.

**Review by jane-intel — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Review by GlebKazantaev — APPROVED** [review]

**Pull request merged**